lib/matplotlib/colors.py

```diff
@@ -3909,3 +4290,34 @@ def from_levels_and_colors(levels, colors, extend='neither'):
 
     norm = BoundaryNorm(levels, ncolors=n_data_colors)
     return cmap, norm
+
+
+def _get_scale_cls_from_str(scale_as_str):
+    """
+    Returns the scale class from a string.
+
+    Used in the creation of norms from a string to ensure a reasonable error
+    in the case where an invalid string is used. This would normally use
+    `_api.check_getitem()`, which would produce the error:
+    'not_a_norm' is not a valid value for norm; supported values are
+    'linear', 'log', 'symlog', 'asinh', 'logit', 'function', 'functionlog'.
+    which is misleading because the norm keyword also accepts `Normalize` objects.
+
+    Parameters
+    ----------
+    scale_as_str : string
+        A string corresponding to a scale
+
+    Returns
+    -------
+    A subclass of ScaleBase.
+
+    """
+    try:
+        scale_cls = scale._scale_mapping[scale_as_str]
+    except KeyError:
+        raise ValueError(
+            f"Invalid norm name {scale_as_str!r}; supported values are "
+            f"{', '.join(scale._scale_mapping)}"
+        ) from None
```

Comment:
Maybe use `_api.check_getitem` which automatically chooses the right keys,
```suggestion
    scale_cls = _api.check_getitem(norm=scale_as_str, scale._scale_mapping)
```
should work.

Reply:
There used to be a comment stating why api.check_getiem is not used here, but not sure where it went. 

In any case, the reason is that the error message produced by api.check_getitem indicates that only stings are valid for the norm keyword (I.e in imshow). In reality, both strings and Normalize objects are valid, which is why the error message here specifies «invalid norm name»

Reply:
But `check_getitem` doesn't say anything about strings, just the keys from the mapping you give it, and this current exception here doesn't say anything about `Normalize` objects either.

Reply:
You might be right that using `_api.check_getitem` is a good idea here, and it has been too long since I looked at this last.
I did some digging around, and this is what I found:

This is separated in a separate function because it is needed by both `MultiNorm` and `Colorizer`.
Initially, this PR was set to change `Colorizer.norm(...)` to use the new function, but I was asked to make the PR more readable by removing all changes not in MultiNorm proper.

In the `Colorizer` class (`colorizer.py`) we currently have:
```py

    @norm.setter
    def norm(self, norm):
        _api.check_isinstance((colors.Norm, str, None), norm=norm)
        if norm is None:
            norm = colors.Normalize()
        elif isinstance(norm, str):
            try:
                scale_cls = scale._scale_mapping[norm]
            except KeyError:
                raise ValueError(
                    "Invalid norm str name; the following values are "
                    f"supported: {', '.join(scale._scale_mapping)}"
                ) from None
            norm = _auto_norm_from_scale(scale_cls)()
        ...
```
Which I believe is syntax inherited from `ScalarMappable`.

Following your comment above, this could be changed to:
```py
    @norm.setter
    def norm(self, norm):
        _api.check_isinstance((colors.Norm, str, None), norm=norm)
        if norm is None:
            norm = colors.Normalize()
        elif isinstance(norm, str):
            scale_cls = _api.check_getitem(scale._scale_mapping, norm=norm)
            norm = _auto_norm_from_scale(scale_cls)()
        ...
 ```
This will change the error of
```py
plt.imshow(np.random.random((3,4)), norm='loG')
```
from 
`ValueError: Invalid norm str name; the following values are supported: linear, log, symlog, asinh, logit, function, functionlog`
to 
`ValueError: 'loG' is not a valid value for norm. Did you mean one of ['log', 'logit']?`
Which sounds like a good change to me, and does not cause any tests to fail on my machine.

---

With this context, I believe we should remove this function and either:
1. 
Repeat the lines
```py

            scale_cls = _api.check_getitem(scale._scale_mapping, norm=norm)
            norm = _auto_norm_from_scale(scale_cls)()
```
in both `Colorizer` and `MultiNorm`.

or 
2. 
Make a function 
```py
def _get_norm_cls_from_str(scale_as_str):
    scale_cls = _api.check_getitem(scale._scale_mapping, norm=scale_as_str)
    norm = colorizer._auto_norm_from_scale(scale_cls)()
    return norm
```
That perform both lines (rather than `_get_scale_cls_from_str(...)` which performed only the first).

I suggest we go with 1. for now, as it allows us to easily finish this PR. 
I have therefore removed `_get_scale_cls_from_str(...)` from this PR.
I have **not** made any changes to `Colorizer`, as that can be done in a separate, independent PR.

Let me know if you agree @QuLogic